### ISSUE TITLE
[Warrior] Autoattack is bugged after using crowd control abilities on your focus target

### ISSUE THREAD

**Issue by PanofskyII** [dalaranwow/dalaran-wow#5851]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
It's quite common for arms warriors (and for every other class/spec in the game) who like to pvp to use many "focus macro". For example: you are doing a 2vs2 arena match against another mdps+healer combo. You try to kill the mdps (aka: your main target)  and you put the opposite healer on focus. When the opposite healer begins to cast a certain healing spell, you use your Charge ability to stop the cast and prevent the heal. In order to do that you can use a macro like this:
#showtooltip Charge
/cast [stance:1,target=focus] Charge; [stance:2/3] Battle Stance

When you use a focus macro on Algalon realm (from my tests on target dummies it happens not only with Charge but with every other ability, like Intercept, Intimidating Shout, Shield Bash, Disarm etc etc) the autoattack gets bugged. How? You will not be able to autottack your main target anymore, instead you will be able to autoattack only your focus target. Which means that a warrior cannot build rage from his autoattacks anymore but only from incoming damage. And a secondary effect is that you may accidentaly autoattack your focus target, breaking a crowd control ability applied to him (like Fear, Hex or Polymorph), if you move nearby. This bug happens even if I have a macro for every single ability I use with the /startattack line added to it. For example, my Mortal Strike macro:
#showtooltip Mortal Strike
/startattack
/cast Mortal Strike

**Expected behaviour**: Tell us what should happen instead
When you use an ability on your focus target, you should be able to autoattack your previous/main target.
**Steps to reproduce the problem**:

1. Move to a location where you can find two target dummies placed one next to the other. 
2. Choose which target dummy should be your main target and which one should be your focus target.
3. Proceed to autoattack your main target.
4. Charge your focus target with a macro similar to the one I've written above.
5. Move back to your main target and check if you are able to autoattack it.
6. Move back to your focus target and you will see that your weapon swing triggers on it.

**Include proofs for this behaviour**
On this video, taken from wotlk retail, you can see a warrior using crowd control abilities on his focus target and then his autoattack proccing on his main target. For example, at 05:24 he charges the druid who is casting cyclone (focus), he gets back to the rogue (main) and few seconds later you can clearly see the autoattack (white damage) proc on him:
https://www.youtube.com/watch?v=MgmooQsclrs

If needed, I may try to record a video of myself using the focus macro on target dummies. 

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

You can research this on several pages like https://wowgaming.altervista.org/aowow/ or https://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by Aridam** [comment]
https://github.com/dalaranwow/dalaran-wow/issues/5457 -- my original report here.
Been "fixed" on their PTR or whatever for nearly two years now.

Just give up -- development is dead here.

**Comment by PanofskyII** [comment]
Well, it's quite sad because this bug make warriors almost unplayable in pvp (especially in arena) and if they want to revive pvp they should fix things like this. 
On a side note, an hunter who plays in my same guild, after I talked to him about it, said that maybe this bugs affects also the other classes. When he uses a misdirect focus macro in raid his autoattack gets bugged and he loses a lot of dps.

**Comment by Aridam** [comment]
The only way to get around this is to integrate `/stopattack`.

Example macro I use in any server that has this bugged:
`#showtooltip
/cancelaura Bladestorm
/startattack [noform:2,nomod:alt]
/stopattack [mod:alt]
/cast [form:1,@mouseover,harm,nodead][form:1]Charge;[form:3,@mouseover,harm,nodead][form:3]Intercept;[form:2,@mouseover,help,nodead][form:2]Intervene`

Got `/stopattack` on a modifier key in case I'm doing any Mouseover Charge/Intercept, because this bug affects Mouseover as well as Focus casts.

You either do something like this, or just have to suffer through because devs are lazy and don't bother or care.

**Comment by PanofskyII** [comment]
I've tried to add the /stopattack line to my Charge focus macro and it seems to work. When I go back to my main target after using one of my main abilities with the /startattack line the autoattack is fixed again. Thank you, Aridam. I may reconsider playing Arms pvp now.
